In [19]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score
from sklearn.metrics import classification_report
from sklearn.svm import SVC, LinearSVC
from sklearn.neighbors import KNeighborsClassifier

data = pd.read_csv('avocado.csv', index_col = [0])


In [20]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 18249 entries, 0 to 11
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Date          18249 non-null  object 
 1   AveragePrice  18249 non-null  float64
 2   Total Volume  18249 non-null  float64
 3   4046          18249 non-null  float64
 4   4225          18249 non-null  float64
 5   4770          18249 non-null  float64
 6   Total Bags    18249 non-null  float64
 7   Small Bags    18249 non-null  float64
 8   Large Bags    18249 non-null  float64
 9   XLarge Bags   18249 non-null  float64
 10  type          18249 non-null  object 
 11  year          18249 non-null  int64  
 12  region        18249 non-null  object 
dtypes: float64(9), int64(1), object(3)
memory usage: 1.9+ MB


In [21]:
data.describe()

AveragePrice  Total Volume          4046          4225          4770  \
count  18249.000000  1.824900e+04  1.824900e+04  1.824900e+04  1.824900e+04   
mean       1.405978  8.506440e+05  2.930084e+05  2.951546e+05  2.283974e+04   
std        0.402677  3.453545e+06  1.264989e+06  1.204120e+06  1.074641e+05   
min        0.440000  8.456000e+01  0.000000e+00  0.000000e+00  0.000000e+00   
25%        1.100000  1.083858e+04  8.540700e+02  3.008780e+03  0.000000e+00   
50%        1.370000  1.073768e+05  8.645300e+03  2.906102e+04  1.849900e+02   
75%        1.660000  4.329623e+05  1.110202e+05  1.502069e+05  6.243420e+03   
max        3.250000  6.250565e+07  2.274362e+07  2.047057e+07  2.546439e+06   

         Total Bags    Small Bags    Large Bags    XLarge Bags          year  
count  1.824900e+04  1.824900e+04  1.824900e+04   18249.000000  18249.000000  
mean   2.396392e+05  1.821947e+05  5.433809e+04    3106.426507   2016.147899  
std    9.862424e+05  7.461785e+05  2.439660e+05   17692.894652      0.939938  
min    0.000000e+00  0.000000e+00  0.000000e+00       0.000000   2015.000000  
25%    5.088640e+03  2.849420e+03  1.274700e+02       0.000000   2015.000000  
50%    3.974383e+04  2.636282e+04  2.647710e+03       0.000000   2016.000000  
75%    1.107834e+05  8.333767e+04  2.202925e+04     132.500000   2017.000000  
max    1.937313e+07  1.338459e+07  5.719097e+06  551693.650000   2018.000000

In [22]:
print(data)

          Date  AveragePrice  Total Volume     4046       4225    4770  \
0   2015-12-27          1.33      64236.62  1036.74   54454.85   48.16   
1   2015-12-20          1.35      54876.98   674.28   44638.81   58.33   
2   2015-12-13          0.93     118220.22   794.70  109149.67  130.50   
3   2015-12-06          1.08      78992.15  1132.00   71976.41   72.58   
4   2015-11-29          1.28      51039.60   941.48   43838.39   75.78   
..         ...           ...           ...      ...        ...     ...   
7   2018-02-04          1.63      17074.83  2046.96    1529.20    0.00   
8   2018-01-28          1.71      13888.04  1191.70    3431.50    0.00   
9   2018-01-21          1.87      13766.76  1191.92    2452.79  727.94   
10  2018-01-14          1.93      16205.22  1527.63    2981.04  727.01   
11  2018-01-07          1.62      17489.58  2894.77    2356.13  224.53   

    Total Bags  Small Bags  Large Bags  XLarge Bags          type  year  \
0      8696.87     8603.62       93.

In [23]:
data = data.drop('Date',axis = 1)
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
data.region = le.fit_transform(data.region)
data.type = le.fit_transform(data.type)
data = data.astype(int)
data.corr()


AveragePrice  Total Volume      4046      4225      4770  \
AveragePrice      1.000000     -0.154704 -0.179887 -0.126938 -0.141933   
Total Volume     -0.154704      1.000000  0.977863  0.974181  0.872202   
4046             -0.179887      0.977863  1.000000  0.926110  0.833389   
4225             -0.126938      0.974181  0.926110  1.000000  0.887855   
4770             -0.141933      0.872202  0.833389  0.887855  1.000000   
Total Bags       -0.140555      0.963047  0.920057  0.905787  0.792314   
Small Bags       -0.135159      0.967238  0.925280  0.916031  0.802733   
Large Bags       -0.149239      0.880640  0.838645  0.810015  0.698471   
XLarge Bags      -0.076810      0.747156  0.699376  0.688807  0.679860   
type              0.423226     -0.232434 -0.225819 -0.232289 -0.210026   
year              0.073560      0.017193  0.003353 -0.009559 -0.036531   
region           -0.020351      0.174176  0.192073  0.145726  0.095252   

              Total Bags  Small Bags  Large Bags  XLarge Bags      type  \
AveragePrice   -0.140555   -0.135159   -0.149239    -0.076810  0.423226   
Total Volume    0.963047    0.967238    0.880640     0.747156 -0.232434   
4046            0.920057    0.925280    0.838645     0.699376 -0.225819   
4225            0.905787    0.916031    0.810015     0.688807 -0.232289   
4770            0.792314    0.802733    0.698471     0.679860 -0.210026   
Total Bags      1.000000    0.994335    0.943009     0.804232 -0.217788   
Small Bags      0.994335    1.000000    0.902589     0.806843 -0.220535   
Large Bags      0.943009    0.902589    1.000000     0.710858 -0.193177   
XLarge Bags     0.804232    0.806843    0.710858     1.000000 -0.175476   
type           -0.217788   -0.220535   -0.193177    -0.175476  1.000000   
year            0.071552    0.063915    0.087891     0.081032 -0.000032   
region          0.175256    0.164702    0.198768     0.082281 -0.000280   

                  year    region  
AveragePrice  0.073560 -0.020351  
Total Volume  0.017193  0.174176  
4046          0.003353  0.192073  
4225         -0.009559  0.145726  
4770         -0.036531  0.095252  
Total Bags    0.071552  0.175256  
Small Bags    0.063915  0.164702  
Large Bags    0.087891  0.198768  
XLarge Bags   0.081032  0.082281  
type         -0.000032 -0.000280  
year          1.000000 -0.000055  
region       -0.000055  1.000000

In [24]:

data =  data.drop('year' , axis = 1)
data =  data.drop('region' , axis =1)
data =  data.drop('XLarge Bags' , axis =1)

data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 18249 entries, 0 to 11
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   AveragePrice  18249 non-null  int64
 1   Total Volume  18249 non-null  int64
 2   4046          18249 non-null  int64
 3   4225          18249 non-null  int64
 4   4770          18249 non-null  int64
 5   Total Bags    18249 non-null  int64
 6   Small Bags    18249 non-null  int64
 7   Large Bags    18249 non-null  int64
 8   type          18249 non-null  int64
dtypes: int64(9)
memory usage: 1.4 MB


In [25]:
X = data.drop('AveragePrice' ,axis =1)
y = data['AveragePrice']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.4, random_state = 0)

In [26]:
#GNB
gnb = GaussianNB()
y_pred = gnb.fit(X_train, y_train).predict(X_test)
acc_gnb = round(gnb.score(X_train, y_train) * 100, 2)
print("Naïve Bayes accuracy is:", acc_gnb)

Naïve Bayes accuracy is: 29.35


In [27]:
##SVM
svc = SVC()
svc.fit(X_train, y_train)
Y_pred = svc.predict(X_test)
acc_svc = round(svc.score(X_train, y_train) * 100, 2)
print("svm accuracy is:", acc_svc)

svm accuracy is: 79.9


In [30]:
##KNN
knn = KNeighborsClassifier(n_neighbors = 3)
knn.fit(X_train, y_train)
Y_pred = knn.predict(X_test)
acc_knn = round(knn.score(X_train, y_train) * 100, 2)
print("KNN accuracy is:",acc_knn)

KNN accuracy is: 90.77


With n_neaighbors = 3, KNN accuracy is %90.77

As I increased n_neighbors the accuracy would decrease so, I decided to stick with n_neighbors = 3. 



In [ ]:
#LinearSVC
LinearSVC = SVC(kernel ='linear', max_iter = 500)
Y_pred = LinearSVC.fit(X_train, y_train).predict(X_test)
acc_LinearSVC = round(LinearSVC.score(X_train, y_train) * 100, 2)
print("Linear SVC accuracy is:", acc_LinearSVC)